lib/ansible/modules/network/aos/aos_device.py

```diff
@@ -0,0 +1,208 @@
+            module.exit_json(changed=False,
+                             name=dev.name,
+                             id=dev.id,
+                             value=dev.value)
+
+        if not module.check_mode:
+            try:
+                dev.approve(location=margs['location'])
+            except (SessionError, SessionRqstError):
+                module.fail_json(msg="Unable to approve device")\
+
+        module.exit_json(changed=True,
+                         name=dev.name,
+                         id=dev.id,
+                         value=dev.value)
+    else:
+        # Check if the device is online
+        if dev.state in ('OOS-READY','IS-READY'):
+            module.exit_json(changed=False,
+                             name=dev.name,
+                             id=dev.id,
+                             value=dev.value)
+        else:
+            module.fail_json(msg="Device is not in 'normal' state")
+
+def aos_device(module):
+    margs = module.params
+
+    try:
+        aos = get_aos_session(module, margs['session'])
+    except:
+        module.fail_json(msg="Unable to login to the AOS server")
+
+    item_name = False
+    item_id = False
+
+    if margs['id'] is not None:
+        item_id = margs['id']
+
+    elif margs['name'] is not None:
+        item_name = margs['name']
+
+    #----------------------------------------------------
+    # Find Object if available based on ID or Name
+    #----------------------------------------------------
+    dev = find_collection_item(aos.Devices,
+                        item_name=item_name,
+                        item_id=item_id)
+
+    if dev.exists is False:
+        module.fail_json(msg="unknown device '%s'" % margs['name'])
+
+    #----------------------------------------------------
+    # Valid device state for reference
+    #----------------------------------------------------
+        # DEVICE_STATE_IS_ACTIVE = 1;
+        # DEVICE_STATE_IS_READY = 2;
+        # DEVICE_STATE_IS_NOCOMMS = 3;
+        # DEVICE_STATE_IS_MAINT = 4;
+        # DEVICE_STATE_IS_REBOOTING = 5;
+        # DEVICE_STATE_OOS_STOCKED = 6;
+        # DEVICE_STATE_OOS_QUARANTINED = 7;
+        # DEVICE_STATE_OOS_READY = 8;
+        # DEVICE_STATE_OOS_NOCOMMS = 9;
+        # DEVICE_STATE_OOS_DECOMM = 10;
+        # DEVICE_STATE_OOS_MAINT = 11;
+        # DEVICE_STATE_OOS_REBOOTING = 12;
+        # DEVICE_STATE_ERROR = 13;
+    #----------------------------------------------------
+    # State == Normal
+    #----------------------------------------------------
+    if margs['state'] == 'normal':
+        aos_device_normal(module, aos, dev)
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            session=dict(required=True, type="dict"),
+            name=dict(required=False),
+            id=dict(required=False),
+            state=dict( choices=['normal'],
+                        default='normal'),
+            approve=dict( required=False ),
```

Comment:
Should this be `type='bool'`, or is it more complex?

If it's more complex then it might be worth expanding the description for this option.